Xamarin.Forms.Platform.Android/AppCompat/FormsFragmentPagerAdapter.cs

```diff
@@ -52,8 +64,28 @@ public override void RestoreState(IParcelable state, ClassLoader loader)
 
 		protected override void Dispose(bool disposing)
 		{
-			if (disposing)
+			if (disposing && !_disposed)
+			{
+				_disposed = true;
+
 				_page = null;
+
+				if (!_fragmentManager.IsDestroyed)
+				{
+					FragmentTransaction transaction = _fragmentManager.BeginTransactionEx();
```

Comment:
What about moving this into the override of *DestroyItem*

I tested this with the Test you have included here and it seems like it works
```C#
		public override void DestroyItem(ViewGroup container, int position, Object @object)
		{
			base.DestroyItem(container, position, @object);

			FragmentManager manager = ((Fragment)@object).FragmentManager;
			FragmentTransaction trans = manager.BeginTransactionEx();
			trans.RemoveEx((Fragment)@object);
			trans.CommitAllowingStateLossEx();
		}
```

Reply:
It'a good idea, my only doubt is related to the transaction management.
It's perhaps possible to trigger a transaction overlap exception if several items are destroyed in a short period of time. 
I hope removing ExecutePendingTransactions for this case will be enough to prevent this issue.